app/src/main/java/ch/epfl/sweng/rps/ui/home/MatchViewModel.kt

```diff
@@ -24,12 +23,12 @@ class MatchViewModel : ViewModel() {
 
     private var _gameService: OfflineGameService? = null
     var currentUserResult: Hand.Result? = null
-    private val uid = ServiceLocator.getInstance().getFirebaseRepository().getCurrentUid()
+    private val uid = FirebaseAuth.getInstance().currentUser?.uid
     fun startOfflineGameService(nEvents: Int, computerPlayer: ComputerPlayer) {
         val gameId = UUID.randomUUID().toString()
         _gameService = OfflineGameService(
             gameId,
-            ServiceLocator.getInstance().getFirebaseRepository(),
+            ServiceLocator.getInstance().repository as FirebaseRepository,
```

Comment:
You cant't cast it that way unless you're surew that it's going to be a FirebaseRepository. You should only do it in the tests or where it's crucially needed.

Reply:
Ok, I've raised the abstraction to Repository